### PR TITLE
Set up tests

### PR DESCRIPTION
**Pull request by namurphy**
This initial pull request is to set up configuration files for tests, and also to make sure that the tests and GitHub apps are working.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Required Packages w/ version
+astropy >= 3.2
+h5py >= 2.9
+matplotlib >= 3.0
+numpy >= 1.15
+scipy >= 1.0
+setuptools >= 41.2
+extension_helpers
```
Comment (StanczakDominik) [inline requirements.txt:8]: Like I said yesterday, `extension_helpers` seems mostly useless for us unless we're include C/Cython code.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+language: python
```
Comment (StanczakDominik) [inline .travis.yml:1]: I've been thinking about whether not to cut travis; as it stands now, it does basically the same thing as azure, except it's configured to also run weekly cron jobs (these also need to be configured in the settings menu over at travis-ci.org !) with stable, numpy-dev and astropy-dev.

It probably is good to have it as a backup for ~~the inevitable future collapse of Microsoft~~ Azure outages, though.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          path: docs/_build/html
+
+      - run:
+          name: "Built documentation is available at:"
+          command: DOCS_URL="${CIRCLE_BUILD_URL}/artifacts/${CIRCLE_NODE_INDEX}/${CIRCLE_WORKING_DIRECTORY/#\~/$HOME}/docs/_build/html/index.html"; echo $DOCS_URL
+
+      - save_cache:
+          key: sample-data-v1
+          paths:
+            - ~/sunpy/data/sample_data
```
Comment (StanczakDominik) [inline .circleci/config.yml:108]: This reminds me that I have branch of administratory/CI changes I split off from the particle tracker PR that I'd been intending to push; I'll do that now, thanks for the reminder. That would clear this file a bunch.

Comment (StanczakDominik) [reply .circleci/config.yml:108]: ... that turned out to be a single block removed.

**Comment by namurphy** [comment]
I'm going to merge this now so that I can make sure the Read the Docs page ends up working correctly.

**Pull request merged**